extensions/projection/README.md

```diff
@@ -0,0 +1,124 @@
+# Projection Extension Specification (`proj`)
+
+**Extension [Maturity Classification](../README.md#extension-maturity): Proposal**
+
+This document explains the fields of the STAC Projection (`proj`) Extension to a STAC Item. Here `proj` is short
+for "projection", and not a reference to the use of the PROJ format for the CRS attribute.
+
+The field names defined herein should be added as fields in the Item's Properties object. 
+
+When specified on an Item, the values are assumed to apply to all Assets in that Item.  For example, an Item may have several related Assets each representing a band for the Item, all of which are in the same native CRS, e.g., a UTM Zone.  However, there may also be assets intended for display, like a preview image or thumbnail, that have been reprojected to a different CRS, e.g., Web Mercator, to better accommodate that use case.  This case of differing projections per Asset is not currently handled by this extension.
+
+## Item Properties fields
+
+| Field Name       | Type                     | Description |
+| ---------------- | ------------------------ | ----------- |
+| proj:epsg        | integer\|null  | **Required** [EPSG code](http://www.epsg-registry.org/) of the datasource |
+| proj:crs         | string \|null  | Recommended. PROJ string representing the Coordinate Reference System (CRS) that the `proj:geometry` and `proj:extent` fields represent |
+| proj:geometry    | Polygon Object | Recommended. Defines the footprint of this item. |
+| proj:extent      | [number]       | Recommended. Bounding box of the item or asset in the native CRS |
+| proj:centroid    | [number]       | Recommended. Coordinates representing the centroid of the item or asset in the native CRS |
+
+**proj:epsg** - A Coordinate Reference System (CRS) is the native reference system (sometimes called a
+'projection') used by the data, and can usually be referenced using an [EPSG code](http://epsg.io).
+If the data does not have a CRS, such as in the case of non-rectified imagery with Ground Control
+Points, proj:epsg should be set to null. It should also be set to null if a CRS exists, but for which
+there is no valid EPSG code.
+
+**proj:crs** - A Coordinate Reference System (CRS) is the native reference system (sometimes called a
+'projection') used by the data. This value is a PROJ string.
+If the data does not have a CRS, such as in the case of non-rectified imagery with Ground Control
+Points, proj:crs should be set to null. It should also be set to null if a CRS exists, but for which
+a PROJ string does not exist.
```

Comment:
Should it be made explicit that the `proj4` format is required? Some people may assume WKT, which has some issues from what I understand.

Another option would be to support multiple formats. This could be done explicitly with a `type` or `format` specifier, or implicitly, like what's [done in RasterFrames](https://github.com/locationtech/rasterframes/blob/3538bda126d885e3daa8b3913a82b3890a78da23/core/src/main/scala/org/locationtech/rasterframes/model/LazyCRS.scala#L53-L57) whereby a single field is used to specify the CRS regardless of format, since each format has a different prefix.

My preference would be to have a single `proj:crs` field that accepts strings of the form:

* `"epsg: 32614"`
* `"+proj=utm +zone=14 +datum=WGS84 +units=m +no_defs"`
* `"PROJCS["WGS_1984_UTM_Zone_14N",GEOGCS["GCS_WGS_1984",DATUM["D_WGS_1984",SPHEROID["WGS_1984",6378137,298.257223563]],PRIMEM["Greenwich",0],UNIT["Degree",0.017453292519943295]],PROJECTION["Transverse_Mercator"],PARAMETER["latitude_of_origin",0],PARAMETER["central_meridian",-99],PARAMETER["scale_factor",0.9996],PARAMETER["false_easting",500000],PARAMETER["false_northing",0],UNIT["Meter",1]]"`

rather than separate fields for two different types. The downside is that you lose stronger typing on the EPSG code value, and I'm unclear how this would affect searching (just document it?).

Reply:
The reasoning was that you'd likely search on the espg field, could search on the PROJ field if that didn't exist, and could use the PROJ field if the epsg field was null. 

I didn't add a WKT option as WKT Original Flavor has some ambiguity and implementation concerns with it (that @geospatialjeff mentions above), and WKT2 isn't yet widely adopted.

Reply:
It's explicit that PROJ is required in the fields table